frontend/packages/ux-editor/src/components/config/editModal/EditOptions/EditCodeList/EditCodeList.tsx

```diff
@@ -73,16 +102,29 @@ export function EditCodeList<T extends SelectionComponentType>({
             </StudioNativeSelect>
           )}
         />
+      ) : (
+        <> </>
```

Comment:
Hele denne komponenten begynner å bli ganske uoversiktlig med alle disse ternære operatorne. Har du vurdert å dele den opp i flere komponenter?

Reply:
Jeg ser for meg følgende som en mulighet:
- Dra StudioFileUploader, Trans og href ut i en parent/wrapper komponent. Navn givning kan være vanskelig for denne.
- Dra FormField og StudioNativeSelect ut i en egen komponent

På denne måten blir de ternære operatorene mer synlige
Hvordan tenker du det er best å splitte den opp?

Reply:
Jeg ville nok tatt utgangspunkt i noe slikt:
```
export function EditCodeList<T extends SelectionComponentType>({
  component,
  handleComponentChange,
}: IGenericEditComponent<T>): ReactElement {
  ...
  return (
    <div>
      <CodeListSelector />
      <StudioFileUploader .../>
      <Trans .../>
    </div>
  );
}

function CodeListSelector(): ReactNode {
  const { org, app } = useStudioEnvironmentParams();
  const { data, status } = useOptionListIdsQuery(org, app);
  switch (status) {
    case "pending": return ...;
    case "error": return ...;
    case "success": return <CodeListSelectorWithData codeListIds={data} />;
  }
}

type CodeListSelectorWithDataProps = {
  codeListIds: string[];
};

function CodeListSelectorWithData({ codeListIds }: CodeListSelectorWithDataProps): ReactNode {
  if (!codeListIds.length) return null;

  return <FormField .../>;
}
```
Har fjerent en del kode for oversiktens skyld.

Reply:
Det ser bra ut, da begynner jeg med dette.